msg/IrlockReport.msg

```diff
@@ -9,3 +9,5 @@ float32 pos_x # tan(theta), where theta is the angle between the target and the
 float32 pos_y # tan(theta), where theta is the angle between the target and the camera center of projection in camera y-axis
 float32 size_x #/** size of target along camera x-axis in units of tan(theta) **/
 float32 size_y #/** size of target along camera y-axis in units of tan(theta) **/
+
+float32[4] q   # Quaternion as recorded by the camera (w, x, y, z order, zero-rotation is 1, 0, 0, 0)
```

Comment:
Can IR lock really report a 3D orientation? Or just yaw?

Reply:
No, but that's the thing. IR Lock is limited, and we should eventually move away from referring to it as the standard way for precision landing.

Everyone I'm coming across today is using some camera like the Realsense, some tag detection algorithm running in ROS, and then sends the coordinates of the detected tag to PX4, such that the tracking and position filtering is done in PX4 and offboard really just performs the detection part.

IR beacons are single points and have no orientation. 

The problem right now is that PX4 either allows you to use the `landing_target_pose` message when sending data, in which case you _must_ run offboard estimation, or you use `irlock_report` messages, which don't even have an orientation: https://github.com/PX4/PX4-Autopilot/blob/6f718cd48d9745e4f8347cb918ccc816860d24cb/src/modules/mavlink/mavlink_receiver.cpp#L2499-L2528

`irlock_report` should be renamed to `visual_marker` or something and include an orientation.

So to answer your question: IR Lock cannot have a 3D orientation. But the way we are using the message these days, it should have one, and the message should be renamed.

Reply:
What's worse is that the mavlink message [LANDING_TARGET](https://mavlink.io/en/messages/common.html#LANDING_TARGET) already "supports" irlock reports and actual landing_target messages all in one.

Perhaps I need to revisit the mavlink message and work with its `frame` field instead of using custom parameters for the orientation the way I did in this PR.